src/openrct2/interface/Viewport.cpp

```diff
@@ -669,44 +669,38 @@ void viewport_update_sprite_follow(rct_window* window)
 void viewport_update_smart_sprite_follow(rct_window* window)
 {
     auto entity = TryGetEntity(window->viewport_smart_follow_sprite);
-    if (entity == nullptr)
+    if (entity == nullptr || entity->Type == EntityType::Null)
     {
         window->viewport_smart_follow_sprite = SPRITE_INDEX_NULL;
         window->viewport_target_sprite = SPRITE_INDEX_NULL;
+        return;
     }
-    else if (entity->Type == EntityType::Guest || entity->Type == EntityType::Staff)
+
+    switch (entity->Type)
     {
-        Peep* peep = TryGetEntity<Peep>(window->viewport_smart_follow_sprite);
-        if (peep == nullptr)
+        case EntityType::Vehicle:
+            viewport_update_smart_vehicle_follow(window);
+            break;
+
+        case EntityType::Guest:
+        case EntityType::Staff:
         {
-            // will never happen
-            window->viewport_smart_follow_sprite = SPRITE_INDEX_NULL;
-            window->viewport_target_sprite = SPRITE_INDEX_NULL;
-            return;
+            Peep* peep = static_cast<Peep*>(entity);
```

Comment:
```suggestion
            if (auto* peep = entity->As<Guest>())
                viewport_update_smart_guest_follow(window, peep);
            else if (auto* staff = entity->As<Staff>())
                viewport_update_smart_staff_follow(window, staff);
```
Could also do it this way

Reply:
Splitting them into their own `EntityType::Guest` and `EntityType::Staff` case, the check for `Is<Guest/Staff>` won't even be necessary anymore.